src/unix/lwt_io.ml

```diff
@@ -1454,6 +1454,41 @@ let with_file ?buffer ?flags ?perm ~mode filename f =
     (fun () -> f ic)
     (fun () -> close ic)
 
+let open_temp_file_with_filename:
+  ?buffer : Lwt_bytes.t ->
+  ?perm : Unix.file_perm ->
+  unit -> (string * output_channel) Lwt.t
+  = fun ?buffer ?perm ->
+    let f = Some [Unix.O_WRONLY; Unix.O_CREAT; Unix.O_TRUNC;
+                  Unix.O_NONBLOCK; Unix.O_EXCL] in
+    let rec helper n =
+      let fname = "lwt_io_temp_file_" ^ string_of_int n in
```

Comment:
Also, I believe this will try to create temporary files in the current directory. We should try in system temporary file directories instead:

- [Unix](https://github.com/ocaml/ocaml/blob/7a315bdb5e66bffdd7cef536373a2e0b753d1b7f/stdlib/filename.ml#L86-L87)
- [Win32](https://github.com/ocaml/ocaml/blob/7a315bdb5e66bffdd7cef536373a2e0b753d1b7f/stdlib/filename.ml#L113-L114)

In particular, the current directory is fairly likely not to have permissions for creating files at all. We don't want to touch it for other reasons as well. For example, it may be monitored by a daemon that restarts builds, etc.

The stdlib also added optional arguments for the caller to control which directory is used: https://caml.inria.fr/pub/docs/manual-ocaml/libref/Filename.html#VALtemp_file. We may want those in the interface as well.

You may be able to use that interface in your tests. It would still be nice to test temporary files in the system directory as well, but you may not have such an easy time scanning it.